src/components/EventsDashboard/EventCard.tsx

```diff
@@ -0,0 +1,141 @@
+        return s !== eventId;
+      });
+      isFavourite = false;
+      setFill(false);
+    } else {
+      newSaved.push(eventId);
+      isFavourite = true;
+      setFill(true);
+    }
+    setSaved(newSaved);
+
+    if (!user) {
+      return;
+    }
+
+    const body = {
+      eventID: id,
+      year,
+      isFavourite,
+    };
+
+    try {
+      await fetchBackend({ endpoint: `/users/favEvent/${user}`, method: "PATCH", data: body });
+    } catch (err) {
+      console.error(err);
+    }
+  };
+
+  const timeStateIndicator = (ev: BiztechEvent) => {
+    const startDate = new Date(ev.startDate);
+    const deadline = new Date(ev.deadline);
+    if (new Date() >= deadline && startDate >= new Date()) {
+      return (
+        <div className="rounded-full font-poppin font-[700] px-3 py-1 text-white bg-secondary-color text-[8px] lg:text-[12px]  flex items-center">
+          COMING UP
+        </div>
+      );
+    } else if (startDate > new Date() && !registered.includes(`${ev.id};${ev.year}`)) {
+      return (
+        <div className="rounded-full px-3 py-1 font-poppins font-[700]  text-white bg-events-coming-up text-[8px] lg:text-[12px] flex items-center">
+          REGISTER BY {`${deadline.getMonth() + 1}/${deadline.getDate()}`}
+        </div>
+      );
+    } else {
+      return <></>;
+    }
+  };
+
+  const registeredIndicator = (ev: BiztechEvent) => {
+    if (registered.includes(`${ev.id};${ev.year}`)) {
+      return <div className="rounded-full font-poppin font-[700] px-3 py-1 text-white bg-black text-[8px] lg:text-[12px]  flex items-center">REGISTERED</div>;
+    } else {
+      return <></>;
+    }
+  };
+
+  return (
+    <>
+      <Link href={`/event/${event.id}/${event.year}`}>
+        <motion.div
+          key={`${event.id + event.year + event.createdAt}`}
+          layout
+          initial={{ y: -10, opacity: 0 }}
+          animate={{ y: 0, opacity: 1 }}
+          exit={{ y: 100, opacity: 0 }}
+          transition={{
+            type: "tween",
+            ease: "easeInOut",
+            duration: 0.3,
+          }}
+          onClick={(e) => e.stopPropagation()}
+        >
+          <div className="bg-events-user-card-bg w-full p-3 rounded-[10px] my-2">
+            <div className="flex flex-row space-x-5 relative">
+              <div className="relative lg:w-[200px] lg:h-[130px] w-[100px] h-[75px] overflow-hidden rounded-lg shrink-0">
+                <Image className="object-cover" fill src={event.imageUrl} alt={event.ename} />
+              </div>
+              <div className="flex flex-col space-y-1 grow">
+                <div className="font-600 text-sm lg:text-[24px] py-0.5 lg:py-2 flex flex-row space-x-3 items-center w-full">
+                  <div>{event.ename}</div>
+                  <div className="grow"></div>
+                  <div className="hidden lg:block">{registeredIndicator(event)}</div>
+                  <div className="hidden lg:block">{timeStateIndicator(event)}</div>
```

Comment:
Do we need these lines? Not sure why they're hidden

Reply:
It's for the mobile design where these indicators are moved to different parts of the card (get rendered based on breakpoint). There might be a workaround using grid for it to be one-lined but I think I'll keep it like this (or we can simply keep one of the mobile/desktop placements as they don't look too different)